src/RobotInterface/include/BipedalLocomotion/RobotInterface/ISensorBridge.h

```diff
@@ -0,0 +1,296 @@
+    std::vector<std::string> magnetometersList; /**< list of magnetometers attached to the bridge */
+    std::vector<std::string> sixAxisForceTorqueSensorsList; /**< list of six axis force torque sensors attached to the bridge */
+    std::vector<std::string> threeAxisForceTorqueSensorsList; /**< list of three axis force torque sensors attached to the bridge */
+    std::vector<std::string> cartesianWrenchesList; /**< list of cartesian wrench streams attached to the bridge */
+    std::vector<std::string> rgbCamerasList; /**< list of rgb cameras attached to the bridge */
+    std::vector<std::string> depthCamerasList; /**< list of depth cameras attached to the bridge */
+};
+
+/**
+ * Sensors bridge interface.
+ */
+class ISensorBridge
+{
+public:
+    using unique_ptr = std::unique_ptr<ISensorBridge>;
+
+    using shared_ptr = std::shared_ptr<ISensorBridge>;
+
+    using weak_ptr = std::weak_ptr<ISensorBridge>;
+
+    /**
+     * Initialize estimator
+     */
+    virtual bool initialize(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler) = 0;
+
+    /**
+     * Get joints list
+     */
+    virtual const std::vector<std::string> getJointsList() = 0;
+
+    /**
+     * Get imu sensors list
+     */
+    virtual const std::vector<std::string> getIMUsList() = 0;
+
+    /**
+     * Get linear accelerometers list
+     */
+    virtual const std::vector<std::string> getLinearAccelerometersList() = 0;
+
+    /**
+     * Get gyroscopes list
+     */
+    virtual const std::vector<std::string> getGyroscopesList() = 0;
+
+    /**
+     * Get orientation sensors list
+     */
+    virtual const std::vector<std::string> getOrientationSensorsList() = 0;
+
+    /**
+     * Get magnetometers sensors list
+     */
+    virtual const std::vector<std::string> getMagnetometersList() = 0;
+
+    /**
+     * Get 6 axis FT sensors list
+     */
+    virtual const std::vector<std::string> getSixAxisForceTorqueSensorsList() = 0;
+
+    /**
+     * Get 6 axis FT sensors list
+     */
+    virtual const std::vector<std::string> getThreeAxisForceTorqueSensorsList() = 0;
+
+   /**
+     * Get cartesian wrenches list
+     */
+    virtual const std::vector<std::string> getCartesianWrenchesList() = 0;
+
+    /**
+     * Get rgb cameras list
+     */
+    virtual const std::vector<std::string> getRGBCamerasList() = 0;
+
+    /**
+     * Get depth cameras list
+     */
+    virtual const std::vector<std::string> getDepthCamerasList() = 0;
+
+    /**
+     * Get joint position
+     * @param[in] jointName name of the joint
+     * @param[out] jointPosition joint position in radians
+     * @param[out] recvTimeInS receive time in seconds
+     * @return true/false in case of success/failure
+     */
+    virtual bool getJointPosition(const std::string& jointName,
+                                  double& jointPosition,
+                                  double& recvTimeInS) = 0;
```

Comment:
> Even if it is a little uglier, a possibility could be to input a double pointer instead of a reference, and default it to a nullptr. Then, inside, there should be a check on whether this pointer is null or not. What do you think?

It's not really clear to me the purpose of this suggestion? Could you explain a bit more?

Reply:
I was thinking at something like:
```c++
virtual bool getSomething(SomeType& something,
                                  double* recvTimeInS = nullptr) = 0;
```
In this way, you can call it like
```c++
SomeType output;
getSomething(output);
```
and if you are also interested in the time
```c++
double time;
SomeType output;
getSomething(output, &time);
```

Another possibility could be also to have both the methods with and without ``recvTimeInS``, where the latter simply calls the former with a dummy ``double``, i.e.
```c++
virtual bool getSomething(SomeType& something,
                                  double& recvTimeInS) = 0;

bool getSomething(SomeType& something,
                                  double& recvTimeInS) //Note, this is not virtual
{
    double dummy;
    return getSomething(something, dummy);
}
```
Does it make sense?

Reply:
Ah, ok! I got your point now. I will make the changes corresponding to the former choice.

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/87/commits/3bbf33a17441c9a8ed86cf5c2528659a2cbb5f5e